### PR TITLE
added apply_patch_status method using Merge strategy

### PR DESCRIPTION
**Pull request by maltesander**

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -84,6 +84,37 @@ impl Client {
             .await
     }
 
+    /// Patches subresource status {} in a given Resource.
```
Comment (lfrancke) [inline src/client.rs:87]: What do the "{}" mean?

File: src/client.rs [hunk 2]
```diff
@@ -84,6 +84,37 @@ impl Client {
...
+    ///   With a strategic merge patch, a list is either replaced or merged depending on its patch strategy.
+    ///   The patch strategy is specified by the value of the patchStrategy key in a field tag in the Kubernetes source code.
+    ///   For example, the Containers field of PodSpec struct has a patchStrategy of merge.
+    pub async fn apply_patch_status<T, S>(&self, resource: &T, status: &S) -> OperatorResult<T>
+    where
+        T: Clone + DeserializeOwned + Meta + Resource,
+        S: Serialize,
+    {
+        let new_status = Patch::Merge(serde_json::json!({
+            "apiVersion": T::API_VERSION,
```
Comment (lfrancke) [inline src/client.rs:106]: Is this really necessary? 
The docs of the `patch_status` method doesn't have it.

**Review by lfrancke — CHANGES_REQUESTED** [review]

File: src/client.rs [hunk 3]
```diff
@@ -84,6 +84,37 @@ impl Client {
...
+    ///   Starting from Kubernetes v1.18, you can enable the Server Side Apply feature so that the control plane tracks managed fields for all newly created objects.
+    /// 2) Json (https://tools.ietf.org/html/rfc6902):
+    ///   This is supported on crate feature jsonpatch only
+    /// 3) Merge (https://tools.ietf.org/html/rfc7386):
+    ///   For example, if you want to update a list you have to specify the complete list and update everything  
+    /// 4) Strategic
+    ///   With a strategic merge patch, a list is either replaced or merged depending on its patch strategy.
+    ///   The patch strategy is specified by the value of the patchStrategy key in a field tag in the Kubernetes source code.
+    ///   For example, the Containers field of PodSpec struct has a patchStrategy of merge.
+    pub async fn apply_patch_status<T, S>(&self, resource: &T, status: &S) -> OperatorResult<T>
```
[the remark this answers is not in the document]
Comment (maltesander) [reply src/client.rs:100]: yeah i tested with apply and then merge, didnt adapt the method name ...will fix that.

File: src/client.rs (hunk 2 above)
Comment (maltesander) [reply src/client.rs:106]: You mean apiVersion? Yes. With spark operator:
source: KubeError { source: Api(ErrorResponse { status: "Failure", message: "Incorrect version specified in apply patch. Specified patch version: , expected: spark.stackable.tech/v1", reason: "BadRequest", code: 400

**Comment by lfrancke** [comment]
Could you merge/fix the conflict? It's probably my documentation :(

File: src/client.rs [hunk 4]
```diff
@@ -111,6 +111,71 @@ impl Client {
...
+    /// Patches subresource status in a given Resource using merge strategy.
+    /// The subresource status must be defined beforehand in the Crd.
+    pub async fn merge_patch_status<T, S>(&self, resource: &T, status: &S) -> OperatorResult<T>
+    where
+        T: Clone + DeserializeOwned + Meta + Resource,
+        S: Serialize,
+    {
+        let new_status = Patch::Merge(serde_json::json!({
+            "apiVersion": T::API_VERSION,
+            "kind": T::KIND,
```
Comment (lfrancke) [inline src/client.rs:145]: I'm 99% certain that these two are only required for the `apply` patch but they don't seem to hurt either....

**Review by lfrancke — DISMISSED** [review]

File: src/client.rs (hunk 4 above)
Comment (maltesander) [reply src/client.rs]: Yes just tested you were right. Removed it! One last commit :)

**Review by lfrancke — APPROVED** [review]

**Pull request merged**